### DISCUSSION TITLE
Deconstructing object conditionally React

### DISCUSSION THREAD

**Discussion by optimalfruitcake** [react-hook-form/react-hook-form#2707] [Q&A] [answered]
I am currently using `react-hook-forms` in a create-react-app. I have componentised the Form usage to make sure that i can re-use it in other location. I am trying to deconstruct an object and i am pretty sure its causing an issue.

**Problem:**

The code worked fine until i introduced `createLoading` check. The `oppLoading` seems to be working fine alone. They return `boolean` **true** or **false** and there is always a default state.

However, with the code below, i get the following error

**TypeError: Cannot destructure property 'ref' of '.for' as it is undefined.**

It doesn't give me much information or where to focus. The error seems to be coming from **react-hook-form**, specifically **onDomRemove.ts** on **line 15**.

      12 | const observer = new MutationObserver((): void => {
      13 |   for (const field of Object.values(fieldsRef.current)) {
      14 |     if (field && field.options) {
    > 15 |       for (const { ref } of field.options) {
         | ^  16 |         if (isDetached(ref)) {
      17 |           removeFieldEventListenerAndRef(field);
      18 |         }

**Where i think the problem is:** FormContainer.tsx

        <form
            {...(props.onSubmit && {
                onSubmit: handleSubmit(onSubmit),
            })}
            {...(props.onBlur && {
                onBlur: handleSubmit(onSubmit),
            })}
            className="opportunity--form"
        >

**Question:** 

Am i deconstructing this right or what would be the right way of deconstructing?

**FormContainer.tsx**: 

    const FormContainer: React.FC<FormContainerProps> = (props) => {
        const { handleSubmit } = useFormContext();
        const { loading: oppLoading } = useSelector((state: ApplicationState) => state.save_opportunity_state);
        const { loading: createLoading } = useSelector((state: ApplicationState) => state.creation_state);
    
        const onSubmit = async (values: any) => {
            try {
                if (props.onBlur) {
                    return await props.onBlur(values);
                }
                if (props.onSubmit) {
                    return await props.onSubmit(values);
                }
            } catch (error) {
                console.error(error);
            }
        };
    
        if (oppLoading || createLoading) {
            return <CircularProgress size="80px" color="primary" />;
        }
    
        return (
            <form
                {...(props.onSubmit && {
                    onSubmit: handleSubmit(onSubmit),
                })}
                {...(props.onBlur && {
                    onBlur: handleSubmit(onSubmit),
                })}
                className="opportunity--form"
            >
                {props.children}
            </form>
        );
    };

**Form.tsx**

    export function Form({ onSubmit, onBlur, ...props }: any) {
        return (
            <FormProvider {...props}>
                <FormContainer onSubmit={onSubmit} onBlur={onBlur}>
                    {props.children}
                </FormContainer>
                <CreateResults />
            </FormProvider>
        );
    }

**Usage:**

            <Form onSubmit={onSubmit} {...form}>
                ...fields here
            </Form>

**Comment by bluebill1049** [comment]
can you please provide a codesandbox for your usage?

**Comment by ikesyo** [comment, marked as answer]
Looks like this was fixed by https://github.com/react-hook-form/react-hook-form/commit/fccece893f01bbf896afdcb08b71bfc2d9240912 (included in v6.5.3 or later).
